### ISSUE TITLE
Tracking issue for 1.0.0 release

### ISSUE THREAD

**Issue by messense** [PyO3/maturin#1489]
## Breaking Changes

* [x] move `RemainingCoreMetadata` over to `pyproject.toml`

## Bug fixes

* [ ] https://github.com/PyO3/maturin/issues/1490
* [x] ~~https://github.com/hwchen/keyring-rs/pull/118~~

## Documentation

* [x] Add migration guide
* [x] Add documentation for `maturin generate-ci`
* [x] Add documentation for maturin related environment variables like the ones added in https://github.com/PyO3/maturin/pull/1483

## Nice to have
* [x] https://github.com/PyO3/maturin/issues/1252

**Comment by konstin** [comment]
I think we should also move `RemainingCoreMetadata` over to pyproject.toml before 1.0. Currently there's a weird split where you need to configure maturin in both Cargo.toml and pyproject.toml; I only realized this when i hit https://www.maturin.rs/project_layout.html#import-rust-as-a-submodule-of-your-project with pep440_rs. Afaik `targets = ` and `data = ` of `[package.metadata.maturin]` are also currently not part of the docs.

**Comment by landhb** [comment]
This should no longer be blocked on `linux-keyutils`. The build has been fixed in the `0.2.3` release.
